src/components/shared/SearchInput.js

```diff
@@ -60,9 +60,9 @@ class SearchInput extends Component<Props> {
     if (this.$input) {
       const input = this.$input;
       input.focus();
-      if (input.value != "") {
-        input.select();
-      }
+
+      const selectStart = this.props.searchType == "quickOpenModal" ? 1 : 0;
+      input.setSelectionRange(selectStart, input.value.length + 1);
```

Comment:
In this case - it's not that we don't want to select the `@` in `@fo` it's that we don't want to select the text `@` if that is the only thing in the input bar.

I'm not a big fan of the search input having logic based on all of the types. e.g. quick open also has types like function search. The reason for this is that when classes know about their consumers, it creates coupling that makes the complexity a function like N x M where M is the number of consumers.

It'd be better if SearchInput exposed an API about it's behavior for instance whether we should select the input text on mount `shouldSelectText`. This way, QuickOpen could have a predicate like `/^(#|@|:)$/`.